packages/compass-indexes/src/components/create-index-modal/create-index-modal.tsx

```diff
@@ -83,6 +83,10 @@ function CreateIndexModal({
   const enableInIndexesGuidanceExp = usePreference('enableIndexesGuidanceExp');
   const showIndexesGuidanceVariant =
     usePreference('showIndexesGuidanceVariant') && enableInIndexesGuidanceExp;
+  if (showIndexesGuidanceVariant && !currentTab) {
+    // If user is in the variant and there has not been a currentTab set, default it to IndexFlow
+    props.onTabClick('IndexFlow');
+  }
```

Comment:
Dispatching actions from a render method is an antipattern, so is dispatching actions to set initial state. If you need your state to have tree distinct values, you should be using redux ability to set initial state to define it (this is already done for the indexes plugin, you can see some initial state being passed to the `createStore` method in `activateIndexesPlugin`). Looking at the logic though, I think you only need to check the same preference flag when tracking the event instead of introducing a third state to the `currentTab` value and needing to account for this change everywhere